### PR TITLE
Backport of Aceinna Sensor support [ESD-1120]

### PR DESCRIPTION
**Pull request by yizow**
Haven't tested for correctness yet, but merge conflicts resolved and seems to build

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.17-exc-poc-41-gedd73ca4
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=968392)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-exc-poc-41-gedd73ca4&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-exc-poc-41-gedd73ca4&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.17-exc-poc-41-gedd73ca4
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.17-exc-poc-41-gedd73ca4)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.17-exc-poc-41-gedd73ca4/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.17-exc-poc-41-gedd73ca4

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.17-exc-poc-12-gaddbaf19
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=969375)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-exc-poc-12-gaddbaf19&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-exc-poc-12-gaddbaf19&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.17-exc-poc-12-gaddbaf19
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.17-exc-poc-12-gaddbaf19)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.17-exc-poc-12-gaddbaf19/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.17-exc-poc-12-gaddbaf19

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.17-exc-poc-12-gfffdcc69
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=969533)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-exc-poc-12-gfffdcc69&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.2.17-exc-poc-12-gfffdcc69&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.17-exc-poc-12-gfffdcc69
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.17-exc-poc-12-gfffdcc69)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.17-exc-poc-12-gfffdcc69/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.17-exc-poc-12-gfffdcc69

File: package/sbp_j1939_bridge/src/sbp.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ */
+
+#include <libsbp/piksi.h>
+#include <libpiksi/sbp_pubsub.h>
+#include <libpiksi/loop.h>
+#include <libpiksi/logging.h>
+
+#include "sbp.h"
+
+#define SBP_SUB_ENDPOINT "ipc:///var/run/sockets/j1939_internal.pub" /* SBP External Out */
```
Comment (silverjam) [inline package/sbp_j1939_bridge/src/sbp.c]: These aren't really SBP, right?

File: package/sbp_j1939_bridge/src/sbp_j1939_bridge.c [hunk 2]
```diff
@@ -26,10 +27,108 @@
...
+struct j1939_message {
+  int (*handler)(const u8 *payload, const u8 payload_length, struct j1939_sbp_state *state);
+  u32 PGN;
+  u8 payload_length;
+};
+
+static int handle_slope_sensor_info_2(const u8 *payload, const u8 payload_length, struct j1939_sbp_state *state)
+{
+  u32 pitch = 0;
+  for (int i = 0; i < 3; i++) {
```
Comment (silverjam) [inline package/sbp_j1939_bridge/src/sbp_j1939_bridge.c]: Magic numbers

File: package/sbp_j1939_bridge/src/sbp_j1939_bridge.c [hunk 3]
```diff
@@ -38,10 +137,22 @@ bool simulator_enabled_watch = false;
 
 pk_endpoint_t *j1939_pub = NULL;
 
-static int notify_simulator_enable_changed(void *context)
+static void parse_PGN(const u8 *data, struct j1939_sbp_state *state)
```
Comment (silverjam) [inline package/sbp_j1939_bridge/src/sbp_j1939_bridge.c]: Suggest `parse_pgn` -- we don't use uppercase in any of our naming convention

File: package/sbp_j1939_bridge/src/sbp_j1939_bridge.c [hunk 4]
```diff
@@ -26,10 +27,108 @@
...
+  u16 angular_velocity[3] = {0, 0, 0}; /* x, y, z axes */
+  for (int i = 0; i < 3; i++) {
+    angular_velocity[i] = payload[i*2] + (payload[i*2 + 1] << 8);
+  }
+
+  (void) payload_length;
+  piksi_log(LOG_ERR, "Angular Rate X, Y, Z : %.2f, %.2f, %.2f", angular_velocity[0] / 128.0 - 250, angular_velocity[1] / 128.0 - 250, angular_velocity[2] / 128.0 - 250);
+
+  state->imu_raw.gyr_x = angular_velocity[0];
+  state->imu_raw.gyr_x = angular_velocity[1];
```
Comment (silverjam) [inline package/sbp_j1939_bridge/src/sbp_j1939_bridge.c]: Should this be `state->imu_raw.gyr_y` and (next line) `state->imu_raw.gyr_z`?

File: package/sbp_j1939_bridge/src/sbp_j1939_bridge.c [hunk 5]
```diff
@@ -26,10 +27,108 @@
...
+  u16 acceleration[3] = {0, 0, 0}; /* x, y, z axes */
+  for (int i = 0; i < 3; i++) {
+    acceleration[i] = payload[i*2] + (payload[i*2 + 1] << 8);
+  }
+
+  (void) payload_length;
+  piksi_log(LOG_ERR, "Acceleration X, Y, Z : %.2f, %.2f, %.2f", (acceleration[0] - 32000) / 100.0, (acceleration[1] - 32000) / 100.0, (acceleration[2] - 32000) / 100.0);
+
+  state->imu_raw.acc_x = acceleration[0];
+  state->imu_raw.acc_x = acceleration[0];
```
Comment (silverjam) [inline package/sbp_j1939_bridge/src/sbp_j1939_bridge.c]: This looks like a copy-paste error too, should probably be [0], [1], [2] and acc_x, acc_y, acc_z?

File: package/sbp_j1939_bridge/src/sbp_j1939_bridge.c [hunk 6]
```diff
@@ -26,10 +27,108 @@
...
+
+  state->imu_raw.gyr_x = angular_velocity[0];
+  state->imu_raw.gyr_x = angular_velocity[1];
+  state->imu_raw.gyr_x = angular_velocity[2];
+
+  return 0;
+}
+
+static int handle_acceleration_sensor(const u8 *payload, const u8 payload_length, struct j1939_sbp_state *state)
+{
```
Comment (silverjam) [inline package/sbp_j1939_bridge/src/sbp_j1939_bridge.c]: More magic numbers in this function.

File: package/sbp_j1939_bridge/src/sbp_j1939_bridge.c [hunk 7]
```diff
@@ -26,10 +27,108 @@
...
+  u16 acc_y;
+  u16 acc_z;
+  u16 gyr_x;
+  u16 gyr_y;
+  u16 gyr_z;
+};
+
 struct j1939_sbp_state {
-  int counter;
+  u32 PGN;
```
Comment (silverjam) [inline package/sbp_j1939_bridge/src/sbp_j1939_bridge.c]: Suggest just `pgn` since we don't use upper case (except for macros).

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.17-exc-poc-13-g741032bc
There was an error using the HITL API to kick off smoke tests for this commit. See https://travis-ci.org/swift-nav/piksi_buildroot/builds/511784174.

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.17-exc-poc-13-g741032bc
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.17-exc-poc-13-g741032bc)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.17-exc-poc-13-g741032bc/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.17-exc-poc-13-g741032bc

**Pull request closed without merging**